subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/maven/M2Installation.groovy

```diff
@@ -98,8 +99,22 @@ class M2Installation implements Action<GradleExecuter> {
     void execute(GradleExecuter gradleExecuter) {
         init()
         gradleExecuter.withUserHomeDir(userHomeDir)
+        // if call `using m2`, then we disable the automatic isolation of m2
+        isolateMavenLocal = false
         if (globalMavenDirectory?.exists()) {
             gradleExecuter.withEnvironmentVars(M2_HOME: globalMavenDirectory.absolutePath)
         }
     }
+
+    void isolateMavenLocalRepo(GradleExecuter gradleExecuter) {
+        gradleExecuter.beforeExecute {
+            if (isolateMavenLocal) {
+                setMavenLocalLocation(gradleExecuter, temporaryFolder.testDirectory.createDir("m2-home-should-not-be-filled"))
+            }
+        }
+    }
+
+    private static void setMavenLocalLocation(GradleExecuter gradleExecuter, File destination) {
+        gradleExecuter.withArgument("-Dmaven.repo.local=${destination?:''}" )
```

Comment:
Why do we need to guard against `null` here? IIUC, `File` should always be non-null, right?

Reply:
It should, yes, I just tend to use this pattern in Groovy rather than seeing `null` at one point :)